src/logger/src/Logger.cpp

```diff
@@ -1,54 +1,24 @@
 #include "logger/Logger.h"
+#include <LockGuard.h>
 #include <bsp/IUserInterface.h>
 #include <cstdarg>
 
-Logger::Logger(LogLevel level, const IUserInterface& ui) : m_ui(ui) {
+Logger::Logger(LogLevel level, const IUserInterface& ui) : m_ui(ui), m_mutex(10) {
     m_logLevel = level;
-    m_semaphore = xSemaphoreCreateBinary();
-
-    if (m_semaphore == NULL) {
-        va_list args;
-        m_ui.printError("Error: Logger semaphore could not be created", args);
-    }
-
-    xSemaphoreGive(m_semaphore);
 }
 
-Logger::~Logger() { vSemaphoreDelete(m_semaphore); }
+LogRet Logger::log(LogLevel level, const char* format, ...) {
+    LockGuard lock = LockGuard(m_mutex);
```

Comment:
Create the lock guard in the if at line 13 , you don't need it if  the levels don't match

Reply:
I will also make the change in HiveMind, as it is the exact same code